test/multiverse/suites/agent_only/agent_run_id_handling_test.rb

```diff
@@ -33,9 +33,15 @@ def test_handles_non_numeric_agent_run_id_on_slow_sql_traces
   end
 
   def simulate_slow_sql_trace
-    in_transaction do
-      agent.sql_sampler.notice_sql('select * from test', 'Database/test/select', nil, 1.5)
+    with_config(:'slow_sql.explain_threshold' => 0.0) do
+      in_transaction do
+        segment = NewRelic::Agent::Tracer.start_datastore_segment
+        # metric name and elapsed args are ignored
+        NewRelic::Agent::Datastores.notice_sql('select * from test', 'Database/test/select', 1.5)
+        segment.finish
+      end
```

Comment:
I found two different strategies for replacing SqlSampler#notice_sql in the tests. The highlighted code above and this approach: 
```
        statement = NewRelic::Agent::Database::Statement.new('SELECT * FROM test')
        metric_name = 'Database/test/select'
        sampler.notice_sql_statement(statement, metric_name, 10)
```

I think the code above might make more sense in areas where transactions are already started. If a transaction isn't explicitly started, I've gone with the second approach.